src/mrack/providers/utils/podman.py

```diff
@@ -81,6 +102,38 @@ async def rm(self, container_id, force=False):  # pylint: disable=invalid-name
         _stdout, _stderr, process = await self._run_podman(args, raise_on_err=False)
         return process.returncode == 0
 
+    async def stop(self, container_id, time=0):
+        """Remove a container."""
+        args = ["stop"]
+        if time:
+            args.append("--time")
+            args.append(time)
+
+        args.append(container_id)
+        _stdout, _stderr, process = await self._run_podman(args, raise_on_err=False)
+        return process.returncode == 0
+
+    async def network(self, network, exists=True):
```

Comment:
This method behaves in a strange way, calling it with a non-existing network with `exists=False`, to ensure it doesn't exist (deleting if it exists), only calls `podman inspect {network}` and that's not clear.

Why not split this into `create_network` and `delete_network` and make them idempotent methods?
Another option: change argument to `action='create'` and actions might be `create` or `remove` as well.

Reply:
yes i can try something and push